pkgs/games/shipwright/default.nix

```diff
@@ -0,0 +1,203 @@
+, libpng
+, bzip2
+, imagemagick
+, StormLib
+, romVariant ? "debug"
+, mqRomVariant ? "debug-mq"
+, requireFile
+, ootRom ? requireFile {
+    name = "oot-${romVariant}.z64";
+    message = ''
+      This nix expression requires that oot-${romVariant}.z64 is already part of the store.
+      To get this file you can dump your Ocarina of Time's cartridge to a file,
+      and add it to the nix store with nix-store --add-fixed sha256 <FILE>.
+
+      The supported variants are:
+       - debug: Ocarina of Time Debug PAL GC (not Master Quest)
+       - pal-gc: Ocarina of Time PAL GameCube (may lead to crashes and instability)
+
+      This is optional if you have imported an Ocarina of Time Master Quest ROM.
+      If so, please set enableOotRom to false and enableOotMqRom to true.
+      If both are checked, Ship of Harkinian will be built with both ROMs.
+      Alternatively, use the `shipwright-mq` package instead of `shipwright`.
+    '';
+    sha256 = {
+      debug = "94bdeb4ab906db112078a902f4477e9712c4fe803c4efb98c7b97c3f950305ab";
+      pal-gc = "f788793d27aac3f8d91be5f242c4134217c615bfddd5c70384521ea2153435d2";
+    }.${romVariant};
+  }
+, ootMqRom ? requireFile {
+    name = "oot-${mqRomVariant}.z64";
+    message = ''
+      This nix expression requires that oot-${mqRomVariant}.z64 is already part of the store.
+      To get this file you can dump your Ocarina of Time Master Quest's cartridge to a file,
+      and add it to the nix store with nix-store --add-fixed sha256 <FILE>.
+
+      The supported variants are:
+       - debug-mq: Ocarina of Time Debug PAL GC MQ (Dungeons will be Master Quest)
+
+      This is optional if you have imported an Ocarina of Time ROM.
+      If so, please set enableOotRom to true and enableOotMqRom to false.
+      If both are checked, Ship of Harkinian will be built with both ROMs.
+      Alternatively, use the `shipwright` package instead of `shipwright-mq`.
+    '';
+    sha256 = {
+      debug-mq = "161a3e8d6e67ecd5d996eb0251a946e6ae2b65089563ab1cd0873eca609b6a5f";
+    }.${mqRomVariant};
+  }
+, enableOotRom ? true
+, enableOotMqRom ? false
+}:
+
+assert (enableOotRom || enableOotMqRom);
+
+stdenv.mkDerivation rec {
+  pname = "shipwright";
+  version = "5.1.3";
+
+  dontUseCmakeConfigure = true;
+
+  src = fetchFromGitHub {
+    owner = "harbourmasters";
+    repo = "shipwright";
+    rev = "776886925172040c8e260afba829aa5c0f448414";
+    hash = "sha256-nDjaFrlBYXgxBW4BvsEI1uJVvXHgNwsro44B2Td85bI=";
+    fetchSubmodules = true;
+  };
+  rev = src.rev;
+
+  enableParallelBuilding = true;
+
+  # Copied from the building guide
+  cmakeFlags = [
+    "DEBUG=0"
+    "OPTFLAGS=-O2"
+  ];
+
+  # Linking fails without this
+  hardeningDisable = [ "all" ];
+
+  # Many warnings are being treated as errors
+  NIX_CFLAGS_COMPILE = [
+    "-Wno-error"
+  ];
+
+  nativeBuildInputs = [ python3 llvmPackages.bintools copyDesktopItems makeWrapper cmake ninja pkg-config lsb-release clang imagemagick ];
```

Comment:
I would really prefer for this to be split with newlines, that makes reading future diffs much nicer. Something like this:
```nix
nativeBuildInputs = [
  python3
  llvmPackages.bintools
  ...
];
```
Same goes for `buildInputs`.